src/wp-includes/class-wp-script-modules.php

```diff
@@ -276,25 +296,65 @@ public function add_hooks() {
 	}
 
 	/**
-	 * Prints the enqueued script modules using script tags with type="module"
-	 * attributes.
+	 * Prints the enqueued script modules in head.
+	 *
+	 * @since 6.9.0
+	 */
+	public function print_head_enqueued_script_modules() {
+		$script_modules            = $this->get_marked_for_enqueue();
+		$sorted_script_modules_ids = $this->get_sorted_dependencies( array_keys( $script_modules ) );
+		foreach ( $sorted_script_modules_ids as $id ) {
+			if ( ! in_array( $id, $this->done, true ) && isset( $script_modules[ $id ] ) && ! $script_modules[ $id ]['in_footer'] ) {
```

Comment:
This can be made more efficient:

```suggestion
		$sorted_script_modules_ids = array_diff( $this->get_sorted_dependencies( array_keys( $script_modules ) ), $this->done );
		foreach ( $sorted_script_modules_ids as $id ) {
			if ( isset( $script_modules[ $id ] ) && ! $script_modules[ $id ]['in_footer'] ) {
```

Reply:
But in looking at `get_sorted_dependencies()` it already excludes anything in `done`.

Reply:
You're right, `get_sorted_dependencies()` does filter out what's already done, so the `!in_array( $id, $this->done, true )` checks in both `print_head_enqueued_script_modules()` and `print_enqueued_script_modules()` are redundant.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/9867/commits/8bc8f69e58652ff8a91caa1c1edeea2f68423c4d.